js/script.js

```diff
@@ -0,0 +1,192 @@
+function giveRoundResult (playMove, compMove) {
+	var valuesMatrix = [[0, 1, -1],
+						[-1, 0, 1],
+						[1, -1, 0]];
+	var roundResult = valuesMatrix[setIndex(compMove)][setIndex(playMove)];
+	writeText(params.playerName + ' played ' + playMove + ', computer played ' + compMove + '... ');
+	switch (roundResult) {
+		case -1: writeText('You lost!<br>'); break;
+		case 0: writeText('A draw!<br>'); break;
+		case 1: writeText('You won! Congratulations!<br>'); break;
+	}
+	return roundResult;
+}
+
+// Modal - end of game.
+function modalGameEnd (text) {
+	document.querySelector('#modal-overlay').classList.add('show');
+	document.querySelector('#modal-gameend').classList.add('show');
+	var endText = document.createElement('p');
+	endText.innerHTML = text;
+	params.final.appendChild(endText);
+	params.newGameActive = false;
+}
+
+// Check for winner of the entire game.
+function gameEnd () {
+	if (params.playerPoints === params.limitOfRounds) {
+		var text = params.playerName + ' won the the entire game!<br>';
+		modalGameEnd(text);
+	}
+	if (params.computerPoints === params.limitOfRounds) {
+		var text = 'Computer won the entire game!<br>';
+		modalGameEnd(text);
+	}
+}
+
+// Count current round number and points of each player.
+function countPoints (roundResult) {
+	switch (roundResult) {
+		case -1: ++params.computerPoints; roundResult = "lost"; break;
+		case 0: roundResult = "draw"; break;
+		case 1: ++params.playerPoints; roundResult = "win"; break;
+	}
+	++params.roundNumber;
+	result.innerHTML = '<br>Round: ' + params.roundNumber + ' Rounds to win: ' + params.limitOfRounds + '<br>' + params.playerName + ' ' + params.playerPoints + ' - ' + params.computerPoints + ' Computer';
+	return roundResult;
+}
+
+// Fill final game result table.
+function fillArray () {
+	var newLine = document.createElement('tr');
+	var activeObject = params.progress[params.progress.length-1];
+	for (var key in activeObject) {
+		var newElement = document.createElement('td');
+		newElement.innerHTML = activeObject[key];
+		newLine.appendChild(newElement);
+	}
+	params.finalArray.appendChild(newLine);
+}
+
+// Begin new round of game.
+function playerMove (dataMove) {
+	if (params.newGameActive) {
+		var playMove = dataMove.target.value;
+		var compMove = computerMove();
+		var roundResult = giveRoundResult(playMove, compMove);
+		roundResult = countPoints(roundResult);
+		params.progress.push({
+			arrRoundNumber: params.roundNumber,
+			arrPlayerMove: playMove,
+			arrComputerMove: compMove,
+			arrRoundResult: roundResult,
+			arrGameResult: params.playerPoints + ' - ' + params.computerPoints
+		});
+		fillArray();
+		gameEnd();
+	}
+	else {
+		writeText('Click New Game first<br>');
+	}
+}
+
+// Clear output and result sections.
+function clearText () {
+	params.output.innerHTML = "";
+	params.result.innerHTML = "";
+	params.final.innerHTML = "";
+	while (params.finalArray.children.length > 1) {
```

Comment:
czy tu potrzebna jest pętla? nie bardzo rozumiem ideę

Reply:
Chodziło o to, żeby zostawić w tablicy nagłówki, czyli pierwszy wiersz (stąd > 1) i jednocześnie usunąć wszystkie wytworzone w czasie danej gry wiersze z zapisem jej przebiegu. Bez tej pętli, kolejne rozgrywki wytwarzałyby wiersze z opisem przebiegu, bez skasowania poprzedniej gry. A samo innerHTML = "" kasowałoby przy okazji nagłówki tabeli.